node/exchange.go

```diff
@@ -0,0 +1,24 @@
+package node
+
+import (
+	"sync"
+)
+
+// Used by header and block exchange service for keeping track of
+// the status of the syncer in them.
+type SyncerStatus struct {
+	started bool
+	m       sync.Mutex
+}
+
+func (syncerStatus *SyncerStatus) setStarted() {
+	syncerStatus.m.Lock()
+	defer syncerStatus.m.Unlock()
+	syncerStatus.started = true
+}
+
+func (syncerStatus *SyncerStatus) getStarted() bool {
```

Comment:
[non-blocking] 
nit: typically `get` is associated with some arbitrary value while `is` is more often associated with a boolean. So `isStarted` would make more sense in how we are using this method.